request-processor/src/tasks.py

```
@@ -141,6 +143,103 @@ def handle_check_file(request_schema, request_data, tmp_dir):
 
+def _download_resource(resource_dir, collection, resource):
+    Path(resource_dir).mkdir(parents=True, exist_ok=True)
+    resource_url = (
+        f"https://files.planning.data.gov.uk/{collection}-collection/"
+        f"collection/resource/{urllib.parse.quote(resource)}"
+    )
+    destination = Path(resource_dir) / resource
+
+    try:
+        workflow.download_file(resource_url, destination)
+    except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)